packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -120,6 +120,63 @@ module.exports = {
     }
   },
 
+  async changePassword(ctx) {
+    const params = _.assign({}, ctx.request.body, ctx.params);
+
+    if (!_.get(ctx, 'state.user.id')) {
+      throw new ApplicationError('You must be authenticated to reset your password');
+    }
+
+    if (
+      params.currentPassword &&
+      params.password &&
+      params.passwordConfirmation &&
+      params.passwordConfirmation === params.password
+    ) {
+      const user = await strapi.entityService.findOne(
+        'plugin::users-permissions.user',
+        ctx.state.user.id
+      );
+
+      if (!user) {
```

Comment:
Do we want to acknowledge the user doesn't exist or should we just throw a generic error here?

I don't think we have ever clarified which side of the that security fence we stand on. :thinking:

Reply:
Yeah makes sense, we had some companies contacting us with sec vulns about saying when a user exists or doesn't (and even looking at the request time since they aren't the same).

It's something we will need to decide on.

Reply:
@alexandrebodin what would you do here?
Should I change it to fail silently and return something like `{ ok: true }`, throw an `ApplicationError` with a more generic message like `Something went wrong` or leave it as is?

Reply:
As you need to be authenticated we already know the user exists (unless concurrency deletion but I don't think it makes sense to consider this at the moment)

So I wouldn't event try and find it here and just get the response from edit instead